dpcpp/base/executor.dp.cpp

```diff
@@ -57,6 +57,7 @@ const std::vector<sycl::device> get_devices(std::string device_type)
         {"accelerator", sycl::info::device_type::accelerator},
         {"all", sycl::info::device_type::all},
         {"cpu", sycl::info::device_type::cpu},
+        {"host", sycl::info::device_type::host},
```

Comment:
Yes, it's a special SYCL "device", which works on the CPU but only sees one core, and has no subgroup at all (not even size 1). I see it as a lightweight version of our OmpExecutor or ReferenceExecutor which can be used when you mainly use a device code, but I'm not sure if that's the real purpose.

Honestly, I don't want to advertise this device type since it's so useless for general use, I put this here only to ensure compatibility and complete coverage.